Tools/build/check_warnings.py

```diff
@@ -125,7 +144,7 @@ def get_unexpected_warnings(
 
 
 def get_unexpected_improvements(
-    files_with_expected_warnings: set[str],
+    files_with_expected_warnings: set[tuple[str, int]],
```

Comment:
Instead of `tuple[str, int]` you can use the namedtuple type name here.

Reply:
And in:
```
def get_unexpected_warnings(
    files_with_expected_warnings: set[tuple[str, int]],
```

Reply:
modified to be `set[FileWarnings]`

Reply:
Also here?
```suggestion
    files_with_expected_warnings: set[FileWarnings],
```